pkg/operator/configobservation/serviceca/observeserviceca.go

```diff
@@ -0,0 +1,76 @@
+	"github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+)
+
+const (
+	serviceCAFilePath      = "/etc/kubernetes/static-pod-resources/configmaps/service-ca/service-ca.crt"
+	serviceCAConfigMapName = "service-ca"
+)
+
+// ObserveServiceCA fills in serviceServingCert.CertFile with the path for a configMap containing the service-ca.crt
+func ObserveServiceCA(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	errs := []error{}
+	prevObservedConfig := map[string]interface{}{}
+
+	topLevelServiceCAFilePath := []string{"service-ca", "service-ca.crt"}
+	currentServiceCAFilePath, _, err := unstructured.NestedString(existingConfig, topLevelServiceCAFilePath...)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	if len(currentServiceCAFilePath) > 0 {
+		if err := unstructured.SetNestedField(prevObservedConfig, currentServiceCAFilePath, topLevelServiceCAFilePath...); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	// Sync the configMap in openshift-config-managed to the well-known resting place that corresponds to the service-ca path.
+	// If neither are set, this updates the destination with an empty source, which deletes the destination resource.
+	err = listers.ResourceSyncer().SyncConfigMap(
+		resourcesynccontroller.ResourceLocation{
+			Namespace: operatorclient.TargetNamespace,
+			Name:      serviceCAConfigMapName,
+		},
+		resourcesynccontroller.ResourceLocation{
+			Namespace: operatorclient.GlobalUserSpecifiedConfigNamespace,
+			Name:      serviceCAConfigMapName,
+		},
+	)
+	if err != nil {
+		errs = append(errs, err)
+		return prevObservedConfig, errs
+	}
+
+	observedConfig := map[string]interface{}{}
+
+	// add service-ca configmap mount path to ServiceServingCert if configmap exists
+	ca, err := listers.ConfigMapLister.ConfigMaps(operatorclient.TargetNamespace).Get(serviceCAConfigMapName)
+	if errors.IsNotFound(err) {
+		// do nothing because we aren't going to add a path to a missing configmap
+		return observedConfig, errs
+	}
+	if err != nil {
+		// we had an error, return what we had before and exit. this really shouldn't happen
+		return prevObservedConfig, append(errs, err)
+	}
+	if len(ca.Data) <= 0 {
+		// do nothing because aren't going to add a path to a configmap with no files
+		return observedConfig, errs
+	}
+	// this means we have this configmap and it has values, so wire up the directory
+	if err := unstructured.SetNestedField(observedConfig, serviceCAFilePath, topLevelServiceCAFilePath...); err != nil {
+		recorder.Eventf("ObserveServiceCAConfigMap", "Failed setting serviceCAFile: %v", err)
```

Comment:
this seems like a warning.

Reply:
f83f0b49
(will rebase and squash commits before merge)